src/Compilers/Core/Portable/Operations/IBranchOperationExtensions.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    public static class IBranchOperationExtensions
+    {
+        /// <summary>
+        /// Gets a loop operation that corresponds to the given branch operation.
+        /// </summary>
+        /// <param name="branchOperation">the branch operation for which a corresponding loop is looked up</param>
+        /// <returns>the corresponding loop operation or <c>null</c> in case not found (e.g. no loop syntax or the branch
+        /// belongs switch instead of loop operation)</returns>
+        public static ILoopOperation GetCorrespondingLoop(this IBranchOperation branchOperation)
+        {
+            if (branchOperation.BranchKind != BranchKind.Break && branchOperation.BranchKind != BranchKind.Continue)
+            {
+                throw new InvalidOperationException("Invalid branch kind type. Finding a corresponding loop requires " +
+                    "'break' or 'continue' kinds, but the current branch kind provided is '{branchOperation.Kind}'.");
```

Comment:
We don't use raw strings in exception messages, as they cannot be localized. Please add this string to the `CodeAnalysisResources.resx`, and construct the message using `string.Format`. Note you'll have to make sure to run the build after you add the file to the resx to make sure that all the xlf files get updated (you should see ~13 or so `.xlf` files in the diff).